tmt/frameworks/beakerlib.py

```diff
@@ -33,6 +33,21 @@ def get_environment_variables(
             invocation: 'TestInvocation',
             logger: tmt.log.Logger) -> tmt.utils.Environment:
 
+        # The beakerlib calls the command in this variable in the following way:
+        # $BEAKERLIB_COMMAND_REPORT_RESULT "$testname" "$result" "$logfile" "$score"
+        # - https://github.com/beakerlib/beakerlib/blob/5a85937f557b735f32996eb55cd6b9a33f3fe653/src/testing.sh#L1076
+        #
+        # If we use the `tmt-report-result` value here, the script will not be compatible
+        # with the third `$logfile` positional argument - it will just ignore it because it
+        # accepts the logfile provided only by `-o/--outputFile` option be default.
+        #
+        # The reason the `rhts-report-result` alias is used here is a compatibility layer
+        # implemented by the script itself. If the script gets called with this name, it
+        # will *accept* the third positional argument as a `logfile`.
+        # - https://github.com/teemtee/tmt/blob/e7cf41d1fe5a4dcbb3270758586f41313e9462ec/tmt/steps/execute/scripts/tmt-report-result#L101
+        bkrlib_report_result_cmd = tmt.steps.execute.TMT_REPORT_RESULT_SCRIPT.aliases[1]
+        assert bkrlib_report_result_cmd == "rhts-report-result"
```

Comment:
Changing the order of aliases, for whatever reason, will cause this to explode, and nothing will stop us from doing so.

How about putting `"rhts-report-result"` into a module-level "constant", and just making sure it's one of the aliases?

```
BEAKERLIB_REPORT_RESULT_COMMAND = 'rhts-report-result'

...

if BEAKERLIB_REPORT_RESULT_COMMAND not in tmt.steps.execute.TMT_REPORT_RESULT_SCRIPT.aliases:
    raise GeneralError("Something like Beakerlib framework requires rhts-report-result script to be installed")
```

The dependency on order isn't stable enough, and it'd be nice if frameworks could reserve their own aliases, like this one, but that would be a more complicated patch.

Reply:
Hello @happz , thanks for the hint.

I've changed the code in https://github.com/teemtee/tmt/pull/3372/commits/0f0f860e717f1cb6ec058a1095f0e11387f1c3fc and also added `tmt.steps.execute.TMT_REPORT_RESULT_SCRIPT.source_filename` into the condition. It seems to me it should be probably there. Do you agree?

Reply:
Agreed, unlikely but possible.

Reply:
Ok, marking this as resolved. Thanks!